integration/test_networks.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2016 Canonical Ltd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from integration.testing import IntegrationTestCase
+from pylxd import exceptions
+
+
+class TestNetworks(IntegrationTestCase):
+    """Tests for `Client.networks.`"""
+
+    def test_get(self):
+        """A network is fetched by its name."""
+        name = self.create_network()
+
+        self.addCleanup(self.delete_network, name)
+        network = self.client.networks.get(name)
+
+        self.assertEqual(name, network.name)
+
+    def test_all(self):
+        """All networks are fetched."""
+        name = self.create_network()
+
+        self.addCleanup(self.delete_network, name)
+
+        networks = self.client.networks.all()
+
+        self.assertIn(name, [network.name for network in networks])
+
+    def test_create_default_arguments(self):
+        """A network is created with default arguments"""
+        name = 'eth10'
+        network = self.client.networks.create(name=name)
+        self.addCleanup(self.delete_network, name)
+
+        self.assertEqual(name, network.name)
+        self.assertTrue(network.managed)
+        self.assertEqual('bridge', network.type)
+        self.assertEqual('', network.description)
+
+    def test_create_with_parameters(self):
+        """A network is created with provided arguments"""
+        kwargs = {
+            'name': 'eth10',
+            'config': {
+                'ipv4.address': '10.10.10.1/24',
+                'ipv4.nat': 'true',
+                'ipv6.address': 'none',
+                'ipv6.nat': 'false',
```

Comment:
I'm not quite happy about this - maybe we should accept Python data types here and cast them silently to Go before sending do LXD API?

Reply:
I hadn't noticed this before.  But now you've made me stop and think about it, yes, these ought to be Python types for `'true' -> True`, `'none' -> None` and `'false' -> False`.  However, I'm wondering, surely the API encoding in JSON.  */ checks */  Nope, you're right, they are stings: https://github.com/lxc/lxd/blob/master/shared/api/network.go#L25

I reckon it would be great if pylxd supported both.  In the accepting code, we could do something like:

```python
# assuming config is as above
config_ = {}
for k, v in config.items():
    if v is None:
       v = 'none'
    elif v in (True, False):
        v = str(v).lower()
```

Then, the create() call would accept both form.  Thoughts?